src/ripple/rpc/handlers/AMMInfo.cpp

```diff
@@ -119,7 +122,8 @@ doAMMInfo(RPC::JsonContext& context)
             ammID = sle->getFieldH256(sfAMMID);
         }
 
-        if (issue1.has_value() == ammID.has_value())
+        if ((issue1.has_value() ^ issue2.has_value()) ||
+            (issue1.has_value() == ammID.has_value()))
```

Comment:
Since this isn't a bitwise operation, I'd prefer `!=` instead of exclusive or (yes, I understand they act the same here). Also, if issue1 has a value and issue2 does not, that's a logic error, not a malformed error. I don't object to the check, but I'd assert and return `rpcINTERNAL` on that error.

Reply:
Now that I look, we should probably detect when `params.isMember(jss::asset) != params.isMember(jss::asset2)` and return malformed on that too.

Reply:
Isn't it invalid parameters if one issue is set but not the other? It means that either `asset` or `asset2` is included but not both. `params.isMember(jss::asset) != params.isMember(jss::asset2)` is handled by `issue1.has_value() != issue2.has_value()`.